src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -538,27 +545,131 @@ private void CreateDropTarget()
             }
         }
 
+        /// <summary>
+        /// Ported from https://github.com/chromium/chromium/blob/master/ui/views/win/fullscreen_handler.cc
+        /// </summary>
+        /// <param name="fullscreen">Fullscreen state.</param>
+        private unsafe void MarkFullscreen(bool fullscreen)
+        {
+            if (_taskBarList == IntPtr.Zero)
+            {
+                Guid clsid = ShellIds.TaskBarList;
+                Guid iid = ShellIds.ITaskBarList2;
+
+                int result = CoCreateInstance(ref clsid, IntPtr.Zero, 1, ref iid, out _taskBarList);
```

Comment:
It's probably better to extract it to a separate `TaskBarList` class with `MarkFullscreenWindow` method and cache the instance. So we won't be doing `CoCreateInstance` and `GetDelegateForFunctionPointer` each time on window state change. Will probably also be useful for other stuff later.

Reply:
+

Reply:
done